frame/message-queue/src/mock.rs

```diff
@@ -301,10 +301,15 @@ where
 	ext
 }
 
-/// Run this closure in test externalities.
-pub fn test_closure<R>(f: impl FnOnce() -> R) -> R {
-	let mut ext = new_test_ext::<Test>();
-	ext.execute_with(f)
+/// Run the function pointer inside externalities and asserts the try_state hook at the end.
+pub fn build_and_execute<T: Config>(test: impl FnOnce() -> ())
```

Comment:
@ggwpez this removes a recently added method by you due to a merge conflict. Could you please give this another review?